pkg/plugins/plugins.go

```diff
@@ -0,0 +1,63 @@
+package plugins
+
+// OpenShiftAzureConfig is a new, Red Hat owned type defined in an external project.  It
+// represents a superset of the configuration required for all steps in the plugin process *after*
+// the manifest is validated.
+//
+// OpenShiftAzureConfig is a versioned type.  It is expected that the configuration will change
+// between releases and may require migration steps in the future.  The storage of this type
+// is opaque to the plugins, they only require that it be passed in.  However, it must be retained
+// to allow it to be lifecycled from version to version.
+type OpenShiftAzureConfig struct{}
```

Comment:
I'm wondering if this is actually a `[]byte`.  If it isn't, is it an `interface` which includes methods to marshal and unmarshal?

Reply:
The storage is opaque to the plugin, but the type should be opaque to the caller as well.

Reply:
yes, agree.  Passing in bytes in the new impl